### PR TITLE
Adds the CiviMed Vending Machine

### PR DESCRIPTION
**Pull request by dustylens**
High priced medical supplies for long duration playerbase needs.

## About the PR

Adds:
New vending machine with a high price modifier. Visuals are a green redress of the conventional medical vendor.
Restock crates with unique graphics.
Unique suite of objectively hilarious advertisements.

## Why / Balance

This vending machine is designed to sell topical medications (gauze, ointment, bruise packs and blood packs) at high above normal market values in order to provide a relief valve for convenience versus time for low pop rounds and to provide medical merchants with a price point to operate within.

## How to test

Spawn vending machine.
Check inventory. Make a purchase.
Open machine with screwdriver.
Spawn restock crate. 
Restock vending machine.
Close vending machine.
Check inventory. Make a purchase.
Break the vending machine. I like using an axe.

## Media

civimed

- [ X ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

None.

**Changelog**

### CODE REVIEW

**Comment by dvir001** [comment]
I don't see the use case for this

**Comment by dvir001** [comment]
> > I don't see the use case for this This is the same as putting a food vender on high costs.
> > Something we tried and just made people buy only from food venders
> > Where is this even placed?
> > Why will I not buy from this vend to avoid time on talking to medical?
> > why won't I buy a medical techfab at that point?
> 
> [Cheackraze](https://github.com/Cheackraze) commented [Jul 8, 2024](https://github.com/new-frontiers-14/frontier-station-14/pull/1577#issuecomment-2215299488)
> 
> i dont necessarily want to have cigarettes be a cannonized legal healing method in game like that. The Dan's Soaked Smokes are one thing because its more of a contraband/rare/back alley/sketchy kind of tone to it, and they arent intended as a direct healing method. Otherwise the idea of a more civillian-focused medical vending machine is actually pretty nice because it allows us to reserve the more medical-focused ones to have better items at better prices and to offer items for more doctor/paramedic tailored play

This still didn't answer all the items on the list.

**Comment by dvir001** [comment]
Assuming we are to remove more meds and supply from ships.
Who is this machine for? the person running to buy all the supply from it and leaving it empty?
The only people who are going to restock the machine are the people who are there to mass buy from it, might be a rare case of a janitor getting sent around.
You can make it infinite if you assume its going to be empty all the time but then it's even faster medical printer.
Dying people wont be coming back to a POI for this machine, at that point they already bought the supply beforehand and skipped the medical need of a doctor visit.
Most game injury as of right now can be cleaned with the base medical supply, from this machine.

Money isn't an issue here, it never was, you can even charge 2 times more and it wont change it.

You also need to consider future changes of incoming medical POI and us wanting to remove more medical vending machines around the sector.

You cannot provide a "lop pop fix" that is available to all pop.

The all point of medical gameplay is that there are people needing meds, and people who need medical care, having more medical machines remove this 2 so im not a fan of adding more medical venders outside of a future medical POI.

**Comment by Cheackraze** [comment]
> Assuming we are to remove more meds and supply from ships. Who is this machine for? the person running to buy all the supply from it and leaving it empty? The only people who are going to restock the machine are the people who are there to mass buy from it, might be a rare case of a janitor getting sent around. You can make it infinite if you assume its going to be empty all the time but then it's even faster medical printer. Dying people wont be coming back to a POI for this machine, at that point they already bought the supply beforehand and skipped the medical need of a doctor visit. Most game injury as of right now can be cleaned with the base medical supply, from this machine.
> 
> Money isn't an issue here, it never was, you can even charge 2 times more and it wont change it.
> 
> You also need to consider future changes of incoming medical POI and us wanting to remove more medical vending machines around the sector.
> 
> You cannot provide a "lop pop fix" that is available to all pop.
> 
> The all point of medical gameplay is that there are people needing meds, and people who need medical care, having more medical machines remove this 2 so im not a fan of adding more medical venders outside of a future medical POI.
> 
> This vending machine might have more logic after upstream "just a week away" medical PR.

the overarching idea is that, eventually, we will still want actual doctors to be able to buy important tools and materials for them to be doctors, but we dont want those same tools/overpowered meds available freely to the wider population, so a civi-med vender with just some gauze and very minor things will help people that still want to carry first aid kits, but then we will also be able to stock the official nano-meds with things like epinephrine vials and actual meaningful supplies

**Comment by dvir001** [comment]
> > Assuming we are to remove more meds and supply from ships. Who is this machine for? the person running to buy all the supply from it and leaving it empty? The only people who are going to restock the machine are the people who are there to mass buy from it, might be a rare case of a janitor getting sent around. You can make it infinite if you assume its going to be empty all the time but then it's even faster medical printer. Dying people wont be coming back to a POI for this machine, at that point they already bought the supply beforehand and skipped the medical need of a doctor visit. Most game injury as of right now can be cleaned with the base medical supply, from this machine.
> > Money isn't an issue here, it never was, you can even charge 2 times more and it wont change it.
> > You also need to consider future changes of incoming medical POI and us wanting to remove more medical vending machines around the sector.
> > You cannot provide a "lop pop fix" that is available to all pop.
> > The all point of medical gameplay is that there are people needing meds, and people who need medical care, having more medical machines remove this 2 so im not a fan of adding more medical venders outside of a future medical POI.
> > This vending machine might have more logic after upstream "just a week away" medical PR.
> 
> the overarching idea is that, eventually, we will still want actual doctors to be able to buy important tools and materials for them to be doctors, but we dont want those same tools/overpowered meds available freely to the wider population, so a civi-med vender with just some gauze and very minor things will help people that still want to carry first aid kits, but then we will also be able to stock the official nano-meds with things like epinephrine vials and actual meaningful supplies

We might want to wait for medical poi then?

**Comment by Cheackraze** [comment]
> > > Assuming we are to remove more meds and supply from ships. Who is this machine for? the person running to buy all the supply from it and leaving it empty? The only people who are going to restock the machine are the people who are there to mass buy from it, might be a rare case of a janitor getting sent around. You can make it infinite if you assume its going to be empty all the time but then it's even faster medical printer. Dying people wont be coming back to a POI for this machine, at that point they already bought the supply beforehand and skipped the medical need of a doctor visit. Most game injury as of right now can be cleaned with the base medical supply, from this machine.
> > > Money isn't an issue here, it never was, you can even charge 2 times more and it wont change it.
> > > You also need to consider future changes of incoming medical POI and us wanting to remove more medical vending machines around the sector.
> > > You cannot provide a "lop pop fix" that is available to all pop.
> > > The all point of medical gameplay is that there are people needing meds, and people who need medical care, having more medical machines remove this 2 so im not a fan of adding more medical venders outside of a future medical POI.
> > > This vending machine might have more logic after upstream "just a week away" medical PR.
> > 
> > 
> > the overarching idea is that, eventually, we will still want actual doctors to be able to buy important tools and materials for them to be doctors, but we dont want those same tools/overpowered meds available freely to the wider population, so a civi-med vender with just some gauze and very minor things will help people that still want to carry first aid kits, but then we will also be able to stock the official nano-meds with things like epinephrine vials and actual meaningful supplies
> 
> We might want to wait for medical poi then?

i mean, thats on whoever is wanting to map it. For now i think this is still a decent change, the public POI venders can be swapped for this civi version for now and then for the time being we can keep ships with the nano-meds unless we see people abusing that again

**Comment by dvir001** [comment]
> > > > Assuming we are to remove more meds and supply from ships. Who is this machine for? the person running to buy all the supply from it and leaving it empty? The only people who are going to restock the machine are the people who are there to mass buy from it, might be a rare case of a janitor getting sent around. You can make it infinite if you assume its going to be empty all the time but then it's even faster medical printer. Dying people wont be coming back to a POI for this machine, at that point they already bought the supply beforehand and skipped the medical need of a doctor visit. Most game injury as of right now can be cleaned with the base medical supply, from this machine.
> > > > Money isn't an issue here, it never was, you can even charge 2 times more and it wont change it.
> > > > You also need to consider future changes of incoming medical POI and us wanting to remove more medical vending machines around the sector.
> > > > You cannot provide a "lop pop fix" that is available to all pop.
> > > > The all point of medical gameplay is that there are people needing meds, and people who need medical care, having more medical machines remove this 2 so im not a fan of adding more medical venders outside of a future medical POI.
> > > > This vending machine might have more logic after upstream "just a week away" medical PR.
> > > 
> > > 
> > > the overarching idea is that, eventually, we will still want actual doctors to be able to buy important tools and materials for them to be doctors, but we dont want those same tools/overpowered meds available freely to the wider population, so a civi-med vender with just some gauze and very minor things will help people that still want to carry first aid kits, but then we will also be able to stock the official nano-meds with things like epinephrine vials and actual meaningful supplies
> > 
> > 
> > We might want to wait for medical poi then?
> 
> i mean, thats on whoever is wanting to map it. For now i think this is still a decent change, the public POI venders can be swapped for this civi version for now and then for the time being we can keep ships with the nano-meds unless we see people abusing that again

Alright fuck it, but no restocks in cargo. X_x

**Comment by dvir001** [comment]
Don't add restocks to cargo, we want to avoid them if possible.

File: Resources/Prototypes/_NF/Catalog/Cargo/cargo_vending.yml [hunk 1]
```diff
@@ -87,3 +87,14 @@
...
+- type: cargoProduct
+  id: CrateVendingMachineRestockCiviMed
+  abstract: true
+  icon:
+    sprite: _NF/Objects/Specific/Service/vending_machine_restock.rsi
+    state: base
+  product: CrateVendingMachineRestockCiviMedFilled
+  cost: 100 # Frontier
+  category: cargoproduct-category-name-fun
+  group: market
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Catalog/Cargo/cargo_vending.yml:100]: Remove the listing please

**Review by dvir001 — APPROVED** [review]
Might need to change the fill but its good to go as its right now.

File: Resources/Locale/en-US/_NF/advertisements/vending/civimed.ftl [hunk 2]
```diff
@@ -0,0 +1,9 @@
+advertisement-civimed-1 = MINERS DO NOT DIE!
+advertisement-civimed-2 = We are better than dying in space!
+advertisement-civimed-3 = We're not overcharging you! We just know what you're worth! Pay up!
+advertisement-civimed-4 = Your money or your life! This is absolutely a threat!
+advertisement-civimed-5 = Nine out of ten corpses did NOT stock up on medicine! The last one was an idiot. 
+advertisement-civimed-6 = Packs laden with loot, are often low on supplies.
+advertisement-civimed-7 = Don't get a skin graft! Stock up on corporate graft! Just buy some medicine.
+advertisement-civimed-8 = I'm not a doctor. But you can be! Self medicate the pain away!
+advertisement-civimed-9 = Relative to other hazards our products are SAFE for humanoid consumption!
```
Comment (whatston3) [inline Resources/Locale/en-US/_NF/advertisements/vending/civimed.ftl:9]:
```suggestion
advertisement-civimed-1 = MINERS, DON'T DIE!
advertisement-civimed-2 = You're better than dying in space! Stock up now!
advertisement-civimed-3 = We're not overcharging you, we just know what you're worth! Pay up!
advertisement-civimed-4 = Your money or your life! This is not legally a threat!
advertisement-civimed-5 = Nine out of ten corpses did NOT stock up on medicine! The last one was an idiot.
advertisement-civimed-6 = Packs laden with loot are often low on supplies.
advertisement-civimed-7 = Get gouged, not burned. Patch yourself up!
advertisement-civimed-8 = I'm not a doctor, but you could be! Self-medicate the pain away!
advertisement-civimed-9 = Focus groups show our products are MOSTLY SAFE for consumption!
```
Suggested rewrites.  Not much of a copy writer.

File: Resources/Prototypes/_NF/Catalog/Cargo/cargo_vending.yml [hunk 3]
```diff
@@ -87,3 +87,14 @@
...
+
+- type: cargoProduct
+  id: CrateVendingMachineRestockCiviMed
+  abstract: true
+  icon:
+    sprite: _NF/Objects/Specific/Service/vending_machine_restock.rsi
+    state: base
+  product: CrateVendingMachineRestockCiviMedFilled
+  cost: 100 # Frontier
+  category: cargoproduct-category-name-fun
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Catalog/Cargo/cargo_vending.yml:99]:
```suggestion
  category: cargoproduct-category-name-medical
```

File: Resources/Prototypes/_NF/Entities/Structures/Machines/vending_machines.yml [hunk 4]
```diff
@@ -599,3 +599,34 @@
       shader: unshaded
     - state: panel
       map: ["enum.WiresVisualLayers.MaintenancePanel"]
+
+- type: entity
+  parent: [BaseStructureUnanchorable, BaseStructureIndestructible, VendingMachine]
+  id: VendingMachineCiviMedVend
+  name: CiviMedVend
+  description: CiviMed! Distributing lifesaving meds harder than the rocks you break! CiviMed! Miners DO NOT DIE!
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Structures/Machines/vending_machines.yml:607]:
```suggestion
  description: CiviMed, distributing lifesaving meds harder than the rocks you break! Miners, DON'T DIE!
```

**Pull request merged**